app/lib/actions/candlepin/consumer/attach_subscriptions.rb

```diff
@@ -0,0 +1,25 @@
+module Actions
+  module Candlepin
+    module Consumer
+      class AttachSubscriptions < Candlepin::Abstract
+        input_format do
+          param :uuid, String
+          param :subscriptions, Hash
+        end
+
+        def plan(system, subscriptions)
+          plan_self(:uuid => system.uuid, :subscriptions => subscriptions)
+        end
+
+        def run
+          if input[:subscriptions]
```

Comment:
If you are checking for this parameter (assumption being it may not have been passed in?), do you also need to check for `input[:uuid]` below?